lib/models/addon.js

```diff
@@ -873,15 +877,32 @@ let addonProto = {
     @return {Tree} The return tree has the same contents as the input tree, but is moved so that the `app/styles/` path is preserved.
   */
   treeForStyles(tree) {
+
+    if (!this.name || this.name === 'superFn') {
+      throw new Error('needs to call _super.treeForStyles.call(...)');
```

Comment:
I guess the point is to correctly pass the `this` context, so maybe include that in your exception message? 

```suggestion
      throw new Error('needs to call _super.treeForStyles.call(this, tree)');
```

Reply:
I removed this part, to not require changes on addons